ghcide/bench/lib/Experiments.hs

```diff
@@ -457,28 +445,87 @@ setup = do
 
   return SetupResult{..}
 
---------------------------------------------------------------------------------------------
+setupDocumentContents :: Config -> Session [DocumentPositions]
+setupDocumentContents config =
+        forM (exampleModules $ example config) $ \m -> do
+        doc <- openDoc m "haskell"
+
+        -- Setup the special positions used by the experiments
+        lastLine <- length . T.lines <$> documentContents doc
+        changeDoc doc [TextDocumentContentChangeEvent
+            { _range = Just (Range (Position lastLine 0) (Position lastLine 0))
+            , _rangeLength = Nothing
+            , _text = T.unlines [ "_hygienic = \"hygienic\"" ]
+            }]
+        let
+        -- Points to a string in the target file,
+        -- convenient for hygienic edits
+            stringLiteralP = Position lastLine 15
+
+        -- Find an identifier defined in another file in this project
+        symbols <- getDocumentSymbols doc
+        case symbols of
+            Left [DocumentSymbol{_children = Just (List symbols)}] -> do
+                let endOfImports = case symbols of
+                        DocumentSymbol{_kind = SkModule, _name = "imports", _range } : _ ->
+                            Position (succ $ _line $ _end _range) 4
+                        DocumentSymbol{_range} : _ -> _start _range
+                        [] -> error "Module has no symbols"
+                contents <- documentContents doc
+
+                identifierP <- searchSymbol doc contents endOfImports
+
+                return $ DocumentPositions{..}
+            other ->
+                error $ "symbols: " <> show other
 
--- Parse the max residency and allocations in RTS -s output
-parseMaxResidencyAndAllocations :: String -> (Int, Int)
-parseMaxResidencyAndAllocations input =
-    (f "maximum residency", f "bytes allocated in the heap")
-  where
-    inps = reverse $ lines input
-    f label = case find (label `isInfixOf`) inps of
-        Just l -> read $ filter isDigit $ head $ words l
-        Nothing -> -1
 
-escapeSpaces :: String -> String
-escapeSpaces = map f
-  where
-    f ' ' = '_'
-    f x = x
+
+--------------------------------------------------------------------------------------------
 
 pad :: Int -> String -> String
 pad n [] = replicate n ' '
 pad 0 _ = error "pad"
 pad n (x:xx) = x : pad (n-1) xx
 
-showMB :: Int -> String
-showMB x = show (x `div` 2^(20::Int)) <> "MB"
+-- | Search for a position where:
+--     - get definition works and returns a uri other than this file
+--     - get completions returns a non empty list
+searchSymbol :: TextDocumentIdentifier -> T.Text -> Position -> Session (Maybe Position)
+searchSymbol doc@TextDocumentIdentifier{_uri} fileContents pos = do
+    -- this search is expensive, so we cache the result on disk
+    let cachedPath = fromJust (uriToFilePath _uri) <.> "identifierPosition"
```

Comment:
Is CI setup to preserve `cachedPath`?

Reply:
No, but we only care to preserve it across experiments. For an example E, we run multiple experiments XX.
Expensive here means 20s